channel/contractbackend.go

```diff
@@ -45,6 +45,15 @@ const (
 // create a TxTimedoutError with additional context.
 var errTxTimedOut = errors.New("")
 
+// SharedExpected Nonce is a map of each expected next nonce of all clients.
+var (
+	SharedExpectedNonces map[ChainID]map[common.Address]uint64
+	SharedNonceMtx       map[ChainID]map[common.Address]*sync.Mutex
+)
+
+// SharedMutex controls the reads and writes on the nonceMtx and ecpectedNextNonce of the ContractBackend.
+var SharedMutex = &sync.Mutex{}
```

Comment:
Why do we even have this `SharedMutex` at all when we already have a `SharedNonceMtx`, can we just remove this one?